DoorGate/DoorPresenter.swift

```diff
@@ -0,0 +1,20 @@
+//
+//  DoorPresenter.swift
+//  
+//
+//  Created by Yuliani Noriega on 1/26/18.
+//
+
+struct DoorPresenter {
+    let doorEntity = DoorEntity()
+    
+    mutating func buttonTap(command:DoorCommand) {
+        let state = doorEntity.state.value //easier to read in the if statements below
+        
+        if command == .goingIn && state < DoorState.locked.rawValue {
+            doorEntity.state.value += 1
+        } else if command == .goingOut && state != DoorState.empty.rawValue{
+            doorEntity.state.value -= 1
```

Comment:
Is it necessary to increment or decrement the state? Would it be easier and less prone to error to just set the state to a specific value for open or locked? Is this just to demonstrate that when the threshold is hit, jamming one way then back the other the state change happens?

Reply:
It's my logic for the door state portion, I've included a final else if something were to ever go wrong